File: game-server/src/main/java/org/forfun/mmorpg/game/skill/facade/SkillFacade.java
package org.forfun.mmorpg.game.skill.facade;

import jforgame.socket.share.annotation.MessageRoute;
import org.forfun.mmorpg.framework.eventbus.Subscribe;
import org.forfun.mmorpg.game.Modules;
import org.forfun.mmorpg.game.base.GameContext;
import org.forfun.mmorpg.game.base.MessageUtil;
import org.forfun.mmorpg.game.database.config.domain.ConfigSkill;
import org.forfun.mmorpg.game.database.user.entity.PlayerEnt;
import org.forfun.mmorpg.game.player.event.PlayerLevelUpEvent;
import org.forfun.mmorpg.game.player.event.PlayerLoginEvent;
import org.forfun.mmorpg.game.skill.message.RespPlayerSkills;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;

import java.util.Collection;
import java.util.Collections;

@Controller
@MessageRoute(module = Modules.SKILL)
public class SkillFacade {

    @Subscribe
    public void onPlayerLogin(PlayerLoginEvent loginEvent) {
        PlayerEnt player = loginEvent.getOwner();
    }

    @Subscribe
    public void onPlayerLevelUp(PlayerLevelUpEvent upEvent) {
        PlayerEnt player = upEvent.getOwner();
        int nowLevel = 10;

        Collection<ConfigSkill> skills = GameContext.getDataManager().queryAll(ConfigSkill.class);
        skills.stream().filter(configSkill -> configSkill.getNeedLevel() < nowLevel).forEach(configSkill -> {
            System.err.println("玩家升级，学会新技能->" + configSkill.getName());
        });

        System.err.println("角色" + player.getId() + "登录，准备下发技能列表");
        RespPlayerSkills resp = new RespPlayerSkills();
        resp.setSkills(Collections.singletonList(1));
        MessageUtil.pushMessage(player.getId(), resp);
    }

}

File: game-server/src/main/java/org/forfun/mmorpg/game/skill/facade/SkillFacade.kt
package org.forfun.mmorpg.game.skill.facade

import jforgame.commons.eventbus.Subscribe
import jforgame.socket.share.annotation.MessageRoute
import org.forfun.mmorpg.game.Modules
import org.forfun.mmorpg.game.base.GameContext
import org.forfun.mmorpg.game.base.MessageUtil
import org.forfun.mmorpg.game.database.config.domain.ConfigSkill
import org.forfun.mmorpg.game.database.user.entity.PlayerEnt
import org.forfun.mmorpg.game.player.event.PlayerLevelUpEvent
import org.forfun.mmorpg.game.player.event.PlayerLoginEvent
import org.forfun.mmorpg.game.skill.message.RespPlayerSkills
import org.springframework.stereotype.Controller

@Controller
@MessageRoute(module = Modules.SKILL)
open class SkillFacade {

    @Subscribe
    fun onPlayerLogin(loginEvent: PlayerLoginEvent) {
        val player = loginEvent.owner
    }

    @Subscribe
    fun onPlayerLevelUp(upEvent: PlayerLevelUpEvent) {
        val player = upEvent.owner
        val nowLevel = 10

        val skills = GameContext.getDataManager()?.queryAll(ConfigSkill::class.java)
        skills?.filter { it.needLevel < nowLevel }?.forEach { configSkill ->
            System.err.println("玩家升级，学会新技能->" + configSkill.name)
        }

        System.err.println("角色" + player.id + "登录，准备下发技能列表")
        val resp = RespPlayerSkills()
        resp.skills = listOf(1)
        MessageUtil.pushMessage(player.id, resp)
    }
}
